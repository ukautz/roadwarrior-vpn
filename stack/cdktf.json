{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "74ca6711-f045-4875-b7ed-edf848e011f7",
  "terraformProviders": [
    {
      "name": "digitalocean",
      "version": "~>2.16",
      "source": "digitalocean/digitalocean"
    },
    {
      "name": "hcloud",
      "version": ">=1.32,<2",
      "source": "hetznercloud/hcloud"
    },
    "cloudinit@>=2.2,<3",
    "local"
  ],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}
