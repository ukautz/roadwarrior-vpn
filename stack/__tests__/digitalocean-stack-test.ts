import { Testing } from "cdktf";
import "cdktf/lib/testing/adapters/jest"; // Load types for expect matchers
import * as digitalocean from "../.gen/providers/digitalocean";
import * as path from "path";
import {
  DigitalOceanStack,
  DigitalOceanStackProps,
} from "../lib/digitalocean-stack";

describe("DigitalOcean Stack", () => {
  const newStack = (props?: Partial<DigitalOceanStackProps>) => {
    const app = Testing.app();
    return new DigitalOceanStack(app, "Stack", {
      digitalOceanToken: "the-token",
      sshKeyPath: path.join(__dirname, "fixtures", "dummy.key"),
      vpnClientPublicKey: "pub-key",
      vpnServerPrivateKey: "priv-key",
      ...props,
    });
  };
  it("Renders as expected", () => {
    expect(Testing.synth(newStack())).toMatchSnapshot();
  });
  it("Creates SSH key from disk", () => {
    expect(Testing.synth(newStack())).toHaveResourceWithProperties(
      digitalocean.SshKey,
      {
        public_key: "public key content",
      }
    );
  });
  it("Creates VPN droplet with defaults", () => {
    expect(Testing.synth(newStack())).toHaveResourceWithProperties(
      digitalocean.Droplet,
      {
        name: "vpn-server",
        region: "fra1",
        size: "s-1vcpu-1gb",
        image: "ubuntu-20-04-x64",
      }
    );
  });
});
