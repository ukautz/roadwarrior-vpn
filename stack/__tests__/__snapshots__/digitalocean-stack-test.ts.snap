// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DigitalOcean Stack Renders as expected 1`] = `
"{
  \\"output\\": {
    \\"client-vpn-address\\": {
      \\"value\\": \\"192.168.6.10/24\\"
    },
    \\"server-id\\": {
      \\"value\\": \\"\${digitalocean_droplet.VpnServer.id}\\"
    },
    \\"server-ip\\": {
      \\"value\\": \\"\${digitalocean_droplet.VpnServer.ipv4_address}\\"
    },
    \\"server-status\\": {
      \\"value\\": \\"\${digitalocean_droplet.VpnServer.status}\\"
    },
    \\"server-vpn-network\\": {
      \\"value\\": \\"192.168.6.1/24\\"
    },
    \\"server-vpn-port\\": {
      \\"value\\": \\"51397\\"
    }
  },
  \\"provider\\": {
    \\"cloudinit\\": [
      {}
    ],
    \\"digitalocean\\": [
      {
        \\"token\\": \\"the-token\\"
      }
    ]
  },
  \\"resource\\": {
    \\"cloudinit_config\\": {
      \\"WireguardCloudInit\\": {
        \\"base64_encode\\": false,
        \\"gzip\\": false,
        \\"part\\": [
          {
            \\"content\\": \\"#cloud-config\\\\napt_upgrade: true\\\\npackages:\\\\n  - wireguard\\",
            \\"content_type\\": \\"text/cloud-config\\"
          },
          {
            \\"content\\": \\"#!/bin/bash\\\\n\\\\nmkdir -p /etc/wireguard\\\\ncd /etc/wireguard\\\\n\\\\numask 077\\\\n\\\\ncat > wg0.conf <<- EOF\\\\n[Interface]\\\\nAddress = 192.168.6.1/24\\\\nListenPort = 51397\\\\nPrivateKey = priv-key\\\\nPostUp = iptables -A FORWARD -i %i -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE\\\\nPostDown = iptables -D FORWARD -i %i -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE\\\\n\\\\n[Peer]\\\\nPublicKey = pub-key\\\\nAllowedIPs = 192.168.6.10/24\\\\nEOF\\\\n\\\\numask 022\\\\nsysctl -w net.ipv4.ip_forward=1\\\\necho \\\\\\"net.ipv4.ip_forward = 1\\\\\\" >> /etc/sysctl.d/forwarding.conf\\\\n\\\\nsystemctl enable wg-quick@wg0\\\\nsystemctl start wg-quick@wg0\\",
            \\"content_type\\": \\"text/x-shellscript\\"
          }
        ]
      }
    },
    \\"digitalocean_droplet\\": {
      \\"VpnServer\\": {
        \\"image\\": \\"ubuntu-20-04-x64\\",
        \\"name\\": \\"vpn-server\\",
        \\"region\\": \\"fra1\\",
        \\"size\\": \\"s-1vcpu-1gb\\",
        \\"ssh_keys\\": [
          \\"\${digitalocean_ssh_key.SshKey.id}\\"
        ],
        \\"user_data\\": \\"\${cloudinit_config.WireguardCloudInit.rendered}\\"
      }
    },
    \\"digitalocean_ssh_key\\": {
      \\"SshKey\\": {
        \\"name\\": \\"Stack Key\\",
        \\"public_key\\": \\"public key content\\"
      }
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"cloudinit\\": {
        \\"source\\": \\"cloudinit\\",
        \\"version\\": \\">=2.2,<3\\"
      },
      \\"digitalocean\\": {
        \\"source\\": \\"digitalocean/digitalocean\\",
        \\"version\\": \\"~>2.16\\"
      }
    }
  }
}"
`;
